src/witness/Main.hs

```diff
@@ -5,48 +5,42 @@ module Main where
 
 import Universum
 
-import Mockable (Production (..), runProduction)
 import System.Wlog (logInfo, logWarning)
 
 import qualified Data.ByteString.Char8 as B8
-import Mockable.Concurrent (fork)
+import Disciplina.Async (async)
 import qualified Network.Transport.TCP as TCP
 import Node (NodeAction (..), defaultNodeEnvironment, noReceiveDelay, node, nodeId,
              simpleNodeEndPoint)
 import System.IO (getChar)
 import System.Random (mkStdGen)
 
-import Disciplina.DB (DBType (WitnessDB))
-import Disciplina.Launcher (BasicNodeParams (..), bracketBasicNodeResources, runBasicRealMode)
+import Disciplina.DB (DBParams (..))
 import Disciplina.Listeners (witnessListeners)
 import Disciplina.Messages (serialisePacking)
 import Disciplina.Transport (bracketTransportTCP)
+import Disciplina.Witness (WitnessParams (..), launchWitnessRealMode)
 import Disciplina.Workers (witnessWorkers)
-import Params (WitnessParams (..), getWitnessParams)
+import qualified Params as Params
 
 main :: IO ()
 main = do
-    WitnessParams {..} <- getWitnessParams
-    let basicParams = BasicNodeParams
-            { bnpLoggingParams = wpLogParams
-            , bnpDBType        = WitnessDB
-            , bnpDBPath        = wpDbPath
+    Params.WitnessParams {..} <- Params.getWitnessParams
+    let witnessParams = WitnessParams
+            { wpLoggingParams = wpLogParams
+            , wpDBParams = DBParams{ dbpPath = wpDbPath }
             }
-    runProduction . bracketBasicNodeResources basicParams $
-        \nr -> runBasicRealMode nr $
-          bracketTransportTCP (15000 {-- connection timeout ms--})
-                              (TCP.defaultTCPAddr "127.0.0.1" "10128") $ \transport -> do
-
-            let prng1 = mkStdGen 0
-
+    launchWitnessRealMode witnessParams $ do
             logInfo "Starting node"
-            node (simpleNodeEndPoint transport) (const noReceiveDelay) (const noReceiveDelay)
-                 prng1 serialisePacking (B8.pack "I am node 1") defaultNodeEnvironment $ \node1 ->
-                        NodeAction (witnessListeners . nodeId $ node1) $ \converse -> do
-                            mapM_ (\w -> fork $ w (nodeId node1) [] converse) witnessWorkers
-                            logInfo "Hit return to stop"
-                            _ <- liftIO getChar
-                            logInfo "Stopping node"
+            -- TODO: This networking can't live without Production and Mockables
+            --       so leaving it commented for now
```

Comment:
Btw, if you are using `cardano-sl`'s `Node` and stuff, note that in `develop` it uses `IO` nowadays instead of `Mockable`s.

Reply:
@gromakovsky We don't use it anymore.